classes/ThankYous/ThankYousRepository.php

```diff
@@ -0,0 +1,1121 @@
+	 */
+	private function QueryAddCreatedBetweenFilter(QueryBuilder $query, array $date_range)
+	{
+		$date_range = $this->utility->FormatDateRange($date_range);
+
+		$lower_date = $date_range[0] ?? null;
+		$upper_date = $date_range[1] ?? null;
+
+		if (!isset($lower_date))
+		{
+			throw new InvalidArgumentException("Failed to Add Created Between Filter to Query, Lower Date not found at offset 0");
+		}
+
+		if (!isset($upper_date))
+		{
+			throw new InvalidArgumentException("Failed to Add Created Between Filter to Query, Upper Date not found at offset 1");
+		}
+
+		if (!is_int($lower_date))
+		{
+			throw new InvalidArgumentException("Failed to Add Created Between Filter to Query, Lower Date is not an integer");
+		}
+
+		if (!is_int($upper_date))
+		{
+			throw new InvalidArgumentException("Failed to Add Created Between Filter to Query, Upper Date is not an integer");
+		}
+
+		$query->AddWhereAndClause(self::THANK_YOU_TABLE . ".date_created BETWEEN " . $lower_date . " AND " . $upper_date);
+	}
+
+	private function QueryAddThankedUserFilter(QueryBuilder $query, array $thanked_user_ids)
+	{
+		if (count($thanked_user_ids) === 0)
+		{
+			return;
+		}
+
+		$invalid_user_ids = [];
+		foreach ($thanked_user_ids as $user_id)
+		{
+			if (!is_int($user_id))
+			{
+				$invalid_user_ids[] = $user_id;
+			}
+		}
+
+		if (!empty($invalid_user_ids))
+		{
+			throw new InvalidArgumentException("Failed to Add Thanked User Filter to Query, invalid user IDs for thanked user filter: " . implode(', ', $invalid_user_ids));
+		}
+
+		$query->AddWhereAndClause(self::THANKED_USERS_TABLE . ".user_id IN in:int:thanked_user_ids", $thanked_user_ids);
+	}
+
+	private function QueryAddTagsFilter(QueryBuilder $query, array $tag_ids)
+	{
+		if (count($tag_ids) === 0)
+		{
+			return;
+		}
+
+		$invalid_tag_ids = [];
+		foreach ($tag_ids as $tag_id)
+		{
+			if (!is_int($tag_id))
+			{
+				$invalid_tag_ids[] = $tag_id;
+			}
+		}
+
+		if (!empty($invalid_tag_ids))
+		{
+			throw new InvalidArgumentException("Failed to Add Tagged Filter to Query, invalid Tag IDs for Tag filter: " . implode(', ', $invalid_tag_ids));
+		}
+
+		$query->AddWhereAndClause(self::THANK_YOU_TAGS_TABLE . ".tag_id IN in:int:tag_ids", $tag_ids);
+	}
+
+	private function QueryAddExtranetFilter(QueryBuilder $query, array $extranet_ids, bool $allow_absence = false)
+	{
+		if (count($extranet_ids) === 0)
+		{
+			return;
+		}
+
+		$where             = "(" . self::USER_TABLE . ".ex_area_id IN (";
+		$first_extranet_id = true;
+		foreach ($extranet_ids as $extranet_id)
+		{
+			if (!is_int($extranet_id))
+			{
+				throw new InvalidArgumentException("Failed to Add Extranet Filter to Query, Extranet ID '" . (string) $extranet_id . "' is not an integer");
+			}
+
+			$where             .= $first_extranet_id ? $extranet_id : ", " . $extranet_id;
+			$first_extranet_id = false;
+		}
+		$where .= ")";
+
+		if ($allow_absence)
+		{
+			$where .= " OR " . self::USER_TABLE . ".ex_area_id IS NULL";
+		}
```

Comment:
Perhaps check where `self::USER_TABLE . ".id IS NULL"`. Extranet's should never be null, but just in case that changes in future, it would be better to check the ID for the absolute truth of whether a user was joined or not.

Reply:
Fixed in f96e435e37223443a940c1e216d737b04c4a8d18